test/controllers/users_controller_test.rb

```diff
@@ -32,4 +32,41 @@
       expect(flash[:result_text]).must_equal "User not found!"
     end
   end
+
+  describe "login" do
+    it "can log in an existing user" do
+      # Arrange
+      user_count = User.count
+
+      # Act
+      user = perform_login
+
+      expect(user_count).must_equal User.count
+
+      # Should also test Flash notices
+
+      expect(session[:user_id]).must_equal user.id
+    end
+
+    it "can log in a new user" do
+      # Arrange
+      user = User.new(provider: "github", username: "peter", uid: 987, email: "peter@griffin.com")
+
+      expect {
+        # Act
+        perform_login(user)
+        # Assert
+      }.must_change "User.count", 1
+
+      # Should also test Flash notices
```

Comment:
for sure! i'd throw in expect(flash[:success]).must_equal "Logged in as new user #{user.username}"